### ISSUE TITLE
Addition of mish Activation Function.

### ISSUE THREAD

**Issue by kartikdutt18** [mlpack/mlpack#2156]
Hi everyone, I wanted to contribute to this organization by adding MISH activation function.
It referenced in the paper here: https://arxiv.org/pdf/1908.08681.pdf
According to the paper, 
"Mish demonstrated better results than both Swish and ReLU. For example, during the classification of CIFAR-100 dataset using a Squeeze Excite -18 Network [8]
with Mish resulted in an increase in Top-1 test accuracy by 0.494% and 1.671% as compared to the same network with Swish and ReLU respectively, similarly with MobileNet v2 [9] where the network with Mish had an increase in Top-1 test accuracy by 1.385% over Swish and
0.8702% over ReLU."
Would it be okay if I pursue this issue?
Also this activation function is also available on [tensorflow](https://www.tensorflow.org/addons/api_docs/python/tfa/activations/mish) and pytorch.

**Comment by rcurtin** [comment]
I think it would be just fine to add that; not sure if anyone else has an opinion.

**Comment by zoq** [comment]
Agreed, would be interesting to have an implementation for that activation function.

**Comment by kartikdutt18** [comment]
Thanks, I will start working on it.

**Comment by kartikdutt18** [comment]
Hi @zoq , @rcurtin I have implemented Mish Activation function and I have also tested it [here](https://colab.research.google.com/drive/1x91A7-iIjej_bngTyFNMAtvYxJiQfOE-). I have implemented the same  [here](https://github.com/mlpack/mlpack/pull/2126).
However I already have a PR open regarding another issue so should stash changes of previous PR, close it and send a new PR or should I create a new branch with changes of mish function only?
The changes can be viewed in the PR #2126 . 
Thanks.

**Comment by kartikdutt18** [comment]
As per conversation on IRC, I have created a new PR with only changes corresponding to this issue. #2158 .
Kindly Have a look,
Thanks.

**Comment by kartikdutt18** [comment]
Hi everyone, this issue has been resolved in PR #2158 so I'm closing this issue.

**Comment by rcurtin** [comment]
Thanks @kartikdutt18! :+1:
